### PR TITLE
feat: hide components until they are defined

### PR DESCRIPTION
**Pull request by supproduction**
hides components until they are defined

closes [HRZ-2572](https://spryker.atlassian.net/browse/HRZ-2572)

[HRZ-2572]: https://spryker.atlassian.net/browse/HRZ-2572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tolerants** [comment]
We are going to release it as a feature (minor) - what feature it brings to the customer?

**Comment by supproduction** [comment]
> We are going to release it as a feature (minor) - what feature it brings to the customer?

Fixed implementation and description

File: libs/platform/core/src/orchestration/components/components-loader.ts [hunk 1]
```diff
@@ -152,33 +155,32 @@ export class ComponentsLoader {
...
 
-    return observableType;
+    return extendedClass;
   }
 
   getComponentClass(
     tag: string
   ): (ComponentType & ComponentStatic) | undefined {
-    return this.componentMap.get(tag)?.componentType;
+    return Object.getPrototypeOf(this.componentMap.get(tag)?.extendedClass);
```
Comment (Sinucid) [inline libs/platform/core/src/orchestration/components/components-loader.ts]: `Object.getPrototypeOf` accepts objects only as parameter, but your expression contains optional chaining, that can cause violation.

Comment (Sinucid) [reply libs/platform/core/src/orchestration/components/components-loader.ts]:
```
if (!this.componentMap.get(tag)?.extendedClass) return;
return Object.getPrototypeOf(this.componentMap.get(tag).extendedClass);
```

**Review by Sinucid — APPROVED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**